db/migrate/20190108135546_rename_websocket_to_remote_console.rb

```diff
@@ -0,0 +1,63 @@
+class RenameWebsocketToRemoteConsole < ActiveRecord::Migration[5.0]
+  class SettingsChange < ActiveRecord::Base; end
+
+  class ServerRole < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class MiqWorker < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  def up
+    rename_column :miq_servers, :has_active_websocket, :has_active_remote_console
+
+    say_with_time("Renaming the websocket server role to remote_console") do
+      ServerRole.where(:name => 'websocket').update(:name => 'remote_console', :description => "Remote Consoles")
+    end
+
+    say_with_time("Renaming the /log/level_websocket key to /log/level_remote_console in SettingsChange") do
+      SettingsChange.where(:key => '/log/level_websocket').update(:key => '/log/level_remote_console')
+    end
+
+    say_with_time("Renaming all websocket_worker related SettingsChange keys to remote_console_worker") do
+      SettingsChange.where("key LIKE '/workers/worker_base/websocket_worker/%'").select('id, key').each do |row|
+        SettingsChange.where(:id => row.id).update(:key => row.key.sub('websocket_worker', 'remote_console_worker'))
```

Comment:
Can't we do this without this second `where`?  We already have the row.  Can we remove the `.select` on line 24 and perform the update on the `row`?

Reply:
Oh yes :laughing: I was thinking too much in SQL :sweat_smile: